src/v/cluster/topics_frontend.cc

```diff
@@ -231,6 +232,16 @@ make_error_result(const model::topic_namespace& tp_ns, std::error_code ec) {
     return topic_result(tp_ns, errc::topic_operation_error);
 }
 
+allocation_request make_allocation_request(const topic_configuration& cfg) {
```

Comment:
do we have any on-disk/network compat. issues we need to worry about?

Reply:
great question, luckily the answer is no, we will have to introduce such changes as soon as we will add custom replicas assignment.